src/base/static/libs/maps/index.js

```diff
@@ -332,201 +339,33 @@ class MainMap {
-    //    })
-    //    .on("load", function() {
-    //      self.map.fire("layer:loaded", { id: config.id });
-    //    })
-    //    .on("tileerror", function() {
-    //      self.map.fire("layer:error", { id: config.id });
-    //    });
-    //  self.layers[config.id] = layer;
-    //} else if (config.layers) {
-    //  // If "layers" is present, then we assume that the config
-    //  // references a Leaflet WMS layer.
-    //  // http://leafletjs.com/reference.html#tilelayer-wms
-    //  layer = L.tileLayer
-    //    .wms(config.url, {
-    //      layers: config.layers,
-    //      format: config.format,
-    //      transparent: config.transparent,
-    //      version: config.version,
-    //      crs: L.CRS.EPSG3857,
-    //      // default TileLayer options
-    //      attribution: config.attribution,
-    //      opacity: config.opacity,
-    //      fillColor: config.color,
-    //      weight: config.weight,
-    //      fillOpacity: config.fillOpacity,
-    //    })
-    //    .on("loading", function() {
-    //      self.map.fire("layer:loading", { id: config.id });
-    //    })
-    //    .on("load", function() {
-    //      self.map.fire("layer:loaded", { id: config.id });
-    //    })
-    //    .on("tileerror", function() {
-    //      self.map.fire("layer:error", { id: config.id });
-    //    });
-    //  self.layers[config.id] = layer;
-    //} else {
-    //  // Assume a tile layer
-    //  // TODO: Isn't type=tile for back compatibility
-    //  layer = L.tileLayer(config.url, config)
-    //    .on("loading", function() {
-    //      self.map.fire("layer:loading", { id: config.id });
-    //    })
-    //    .on("load", function() {
-    //      self.map.fire("layer:loaded", { id: config.id });
-    //    })
-    //    .on("tileerror", function() {
-    //      self.map.fire("layer:error", { id: config.id });
-    //    });
-    //  self.layers[config.id] = layer;
-    // }
+    if (config.type === "mapbox-style") {
+      this.map.addMapboxStyle(config.url);
+    } else if (config.type === "raster-tile") {
+      this.layers[config.id] = this.map.createRasterTileLayer(config);
+      this.map.addLayer(this.layers[config.id]);
+    } else if (config.type === "wms") {
+      this.layers[config.id] = this.map.createWMSLayer(config);
+      this.map.addLayer(this.layers[config.id]);
+    } else if (config.type === "wmts") {
+      this.layers[config.id] = this.map.createWMTSLayer(config);
+      this.map.addLayer(this.layers[config.id]);
+    } else if (config.type === "vector-tile") {
+      const vectorLayerGroupConfig = await this.map.createVectorTileLayer(
+        config,
+      );
+      this.layers[config.id] = vectorLayerGroupConfig;
+      this.map.addVectorLayerGroup(this.layers[config.id]);
+    } else if (config.type === "json") {
+      this.layers[config.id] = this.map.createGeoJSONLayer(config);
+      this.map.addGeoJSONLayer(this.layers[config.id], config.geometry_type);
+    }
```

Comment:
To avoid race conditions with the `addVectorLayerGroup` fetch, we should make the following `async/await` refactor when this method is called:

```
 $(Shareabouts).on("visibility", async (evt, id, visible, isBasemap) => {
```
and 

```
 await this.createLayerFromConfig(config);
```